screeps-game-api/src/objects/impls/room_terrain.rs

```diff
@@ -0,0 +1,15 @@
+use {constants::Terrain, objects::RoomTerrain};
+
+impl RoomTerrain {
+    pub fn constructor(room_name: &str) -> Self {
+        js_unwrap!{new Room.Terrain(@{room_name})}
+    }
+
+    pub fn get(&self, x: u32, y: u32) -> Terrain {
+        js_unwrap!{@{self.as_ref()}.get(@{x}, @{y})}
+    }
+
+    pub fn get_raw_buffer(&self) -> Vec<u8> {
+        js_unwrap!(@{self.as_ref()}.getRawBuffer())
```

Comment:
As mentioned in my comment, I think it should be pretty simple to create the `Vec<u8>` ourselves and ask the game to populate it rather than having the game create it and then copy it into WASM.